Fix and improve "displayitem" command, add <server.list_display_items> tag

MusicScore opened the pull request:
Sorry for the vague commit name :(

Tested to work for Spigot 1.12.2, Spigot 1.13.2, and Paper 1.14.4.

MusicScore committed:
Fixed bug, added improvements and new tag

Add <server.list_display_items> tag to get all items created by "displayitem" command

Add EntityHelper#makeItemDisplayOnly(Item) to assist with making items display-only

Fixed strange age error in "displayitem" command

Add "no_gravity" and "permanent" options

Enforce display item location

ghost commented:
Before anyone else makes an approval:

I believe that the `<server.list_display_items>` tag will not register the permanent/temporary display items after a server restart. I will need to save the items and their remaining durations (or no duration) to file to re-register them and obliterate them when necessary.

MusicScore committed:
Multiple improvements to DisplayItem command

Allow recalling permanent display items after server restart

Automatically remove temporary display items on server restart

Add more server tags to get all temporary/permanent display items

Add a safeguard method to ensure that the sets of recognized display items are not bloated with nonexistent item entities

ghost commented:
The above concerns are addressed. Changes were tested and fully functional on a Paper 1.14.4 server.

MusicScore committed:
Fix minor bug in DisplayItemCommand

MusicScore committed:
Add 1.15 NMS method, fix build error
